src/emu/debug/debughlp.cpp

```diff
@@ -35,16 +37,29 @@ struct help_item
 
 
 
+
+/***************************************************************************
+    MACROS
+***************************************************************************/
+
+#define HTML_TRY(f) do { if (f < 0) return false; } while (0)
```

Comment:
Macros should be avoided where possible - they can't be scoped and cause confusing error messages because they work by lexical substitution.  Error handling for one function definitely shouldn't be implemented using a global macro.  Exceptions would work far better.

Reply:
But fprintf doesn't throw exceptions. Should I trow my own, or just close the file and return the regular C way?